data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/ParseTreeParser.java

```diff
@@ -23,17 +25,23 @@
  * Handles interaction with ANTLR generated parser and lexer classes and caches results.
  */
 @Named
+@Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
 class ParseTreeParser implements Parser<ParseTree> {
+    private static final String MISSING_LISTENER_MESSAGE =
```

Comment:
Take it or leave it: a better name for this message maybe: `MISSING_PARSER_ERROR_LISTENER` because there are other types of listeners.